src/rust/src/properties_reader.rs

```diff
@@ -27,6 +27,61 @@ impl FgbFeature {
     pub fn geometry(&self) -> Option<Geometry> {
         self.fbs_feature().geometry()
     }
+
+    fn dimension(&self) -> geo_traits::Dimensions {
+        match (self.header().has_z(), self.header().has_m()) {
+            (true, true) => geo_traits::Dimensions::Xyzm,
+            (true, false) => geo_traits::Dimensions::Xyz,
+            (false, true) => geo_traits::Dimensions::Xym,
+            (false, false) => geo_traits::Dimensions::Xy,
+        }
+    }
+
+    /// Access the underlying geometry, returning an object that implements
+    /// [geo_traits::GeometryTrait].
+    ///
+    /// Note that this method only supports the core geometry types supported by [geo_traits], and
+    /// will panic on curve geometries.
+    pub fn geometry_trait_impl(&self) -> Result<Option<impl geo_traits::GeometryTrait + use<'_>>> {
+        if let Some(geom) = self.geometry() {
+            let dim = self.dimension();
+            let result = match self.header().geometry_type() {
+                GeometryType::Point => crate::geo_trait_impl::Geometry::Point(
+                    crate::geo_trait_impl::Point::new(geom, dim),
+                ),
+                GeometryType::LineString => crate::geo_trait_impl::Geometry::LineString(
+                    crate::geo_trait_impl::LineString::new(geom, dim),
+                ),
+                GeometryType::Polygon => crate::geo_trait_impl::Geometry::Polygon(
+                    crate::geo_trait_impl::Polygon::new(geom, dim),
+                ),
+                GeometryType::MultiPoint => crate::geo_trait_impl::Geometry::MultiPoint(
+                    crate::geo_trait_impl::MultiPoint::new(geom, dim),
+                ),
+                GeometryType::MultiLineString => crate::geo_trait_impl::Geometry::MultiLineString(
+                    crate::geo_trait_impl::MultiLineString::new(geom, dim),
+                ),
+                GeometryType::MultiPolygon => crate::geo_trait_impl::Geometry::MultiPolygon(
+                    crate::geo_trait_impl::MultiPolygon::new(geom, dim),
+                ),
+                GeometryType::Unknown => crate::geo_trait_impl::Geometry::new(geom, dim),
+                GeometryType::GeometryCollection => {
+                    crate::geo_trait_impl::Geometry::GeometryCollection(
+                        crate::geo_trait_impl::GeometryCollection::new(geom, dim),
+                    )
+                }
+                geom_type => {
+                    return Err(GeozeroError::Geometry(format!(
```

Comment:
Should this return `flatgeobuf::Error` instead of `GeozeroError`? I'd think it should, however none of the `flatgeobuf::Error` variants are obvious to use:
```
pub enum Error {
    MissingMagicBytes,
    NoIndex,
    HttpClient([HttpError](https://docs.rs/http-range-client/0.8.0/x86_64-unknown-linux-gnu/http_range_client/error/enum.HttpError.html)),
    IllegalHeaderSize([usize](https://doc.rust-lang.org/nightly/std/primitive.usize.html)),
    InvalidFlatbuffer([InvalidFlatbuffer](https://docs.rs/flatbuffers/24.3.25/x86_64-unknown-linux-gnu/flatbuffers/verifier/enum.InvalidFlatbuffer.html)),
    IO([Error](https://doc.rust-lang.org/nightly/std/io/error/struct.Error.html)),
}
```

Reply:
Yes, we should add an FGB Error for that.